test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs

```diff
@@ -515,28 +520,207 @@ public void ParseFilterEnumTypesWrongCast3()
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberIntegralValueWithSingleQuotes()
+        {
+            // Arrange
+            int enumValue = 1;
+            string filterQuery = "'1' in Colors";
+            string expectedLiteral = "'Red'";
+
+            IEdmEnumType colorType = this.GetIEdmType<IEdmEnumType>("NS.Color");
+            bool success = colorType.TryParse(enumValue, out IEdmEnumMember enumMember);
+
+            // Act
+            FilterClause filterQueryNode = ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+            SingleValueNode expression = filterQueryNode.Expression;
+
+            // Assert
+            Assert.True(success);
+            InNode inNode = expression.ShouldBeInNode();
+            ConstantNode leftNode = Assert.IsType<ConstantNode>(inNode.Left);
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetIEdmProperty("Colors"));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }
+
+        [Fact]
+        public void ParseFilterWithInOperatorWithEnumsMemberIntegralValueWithoutSingleQuotes()
+        {
+            // Arrange
+            int enumValue = 3;
+            string filterQuery = "3 in Colors";
+            string expectedLiteral = "'Blue'";
+
+            IEdmEnumType colorType = this.GetIEdmType<IEdmEnumType>("NS.Color");
+            bool success = colorType.TryParse(enumValue, out IEdmEnumMember enumMember);
+
+            // Act
+            FilterClause filterQueryNode = ParseFilter(filterQuery, this.userModel, this.entityType, this.entitySet);
+            SingleValueNode expression = filterQueryNode.Expression;
+
+            // Assert
+            Assert.True(success);
+            InNode inNode = expression.ShouldBeInNode();
+            ConstantNode leftNode = Assert.IsType<ConstantNode>(inNode.Left);
+            leftNode.ShouldBeEnumNode(colorType, enumMember.Name);
+
+            Assert.True(leftNode.TypeReference.IsEnum());
+            Assert.Equal(enumMember.Name, leftNode.Value.ToString());
+            Assert.Equal(expectedLiteral, leftNode.LiteralText);
+
+            CollectionPropertyAccessNode rightNode = Assert.IsType<CollectionPropertyAccessNode>(inNode.Right);
+            rightNode.ShouldBeCollectionPropertyAccessQueryNode(this.GetIEdmProperty("Colors"));
+            Assert.Equal(colorType, rightNode.ItemType.Definition);
+        }

```

Comment:
You could instead use a single parameterized test that verifies that expressing the filter in the different forms results into the same outcome, i.e.:

```csharp
[Theory]
[InlineData("NS.Color'Green'")]
[InlineData("'Green'")]
[InlineData("'2'")]
[InlineData("2")]
public void ParseFilterWithInOperatorWithEnums(string filterOptionValue)
{
    // Arrange
    string filterQuery = $"{filterOptionValue} in Colors";

    // ...
}
```

Reply:
@gathogojr Thanks for pointing this out. I have resolve this and push the changes. [test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs#L583](https://github.com/OData/odata.net/blob/fix/2371-allow-filter-stringliterals-in-enum-collection/test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EnumFilterFunctionalTests.cs#L583)